### ISSUE TITLE
Let users receive notifications when documents uploaded matching certain criteria

### ISSUE THREAD

**Issue by reefdog** [documentcloud/documentcloud#362]
Feature request after talking with the folks from [CPDP](http://cpdb.co/), who collect and upload complaint reports but also have to keep an eye on the public corpus for other people uploading complaint reports. Currently they just schedule a daily DocumentCloud search for all the CR IDs they're keeping an eye out for.

It would be *much nicer* if, within DC, they could just create a notification. "When a document is uploaded with a `title:CR*12345`, notify us." I can imagine this being useful for others as well.

**Comment by stefangeorg** [comment]
Thanks for creating this Issue, watching this and will be able to help test when it is available. I would, in an ideal world provide a URL endpoint which receives JSON data as a POST request.

**Comment by knowtheory** [comment]
@reefdog this should be titled `Let users receive notifications when documents uploaded matching certain criteria`.  Additionally there needs to be a companion ticket for this one (if there isn't one already) allowing journalists to make things accessible but not public (as vimeo does for example).

Additionally, my vast preference for this is for a feature such as this to be implemented as webhooks so that the interaction is push not pull on our end.

**Comment by reefdog** [comment]
@knowtheory It *is* titled `Let users receive notifications when documents uploaded matching certain criteria`. And agree with both webhooks and companion ticket, though on the latter do you mean Vimeo (password-protected videos) or YouTube (openly-accessible videos that don't show up in any public search or listing, you have to have the URL to access them)? I would think the latter.

**Comment by knowtheory** [comment]
hah.  That was supposed to be

> Let users receive notifications when documents matching certain criteria **are made public**

**Comment by stefangeorg** [comment]
For our project (CPDB), we wouldn't mind getting a push every time a document is updated (whether made public or not), as we would then update our reference in our database.

**Comment by reefdog** [comment]
@knowtheory Ah, I get your point, which I'll address in a sec.

@stefangeorg Ted clarified "are made public" because docs are uploaded private by default, and someone from Organization A shouldn't be notified that a matching document was uploaded by Organization B; instead, they should be notified when Organization B makes that document public.

But Ted, while that's an important point re: interorganization notifications, I'd still expect a notification to be fired from private docs within my organization. So basically: notifications should obey access controls, and "uploaded" should be replaced with "becomes visible to the notification author".

**Comment by knowtheory** [comment]
Right, it was not clear to me whether this was intended as a mechanism to track what _other_ organizations are posting, or whether it was intended to keep tabs on an organization's documents' lifecycles.
